components/settings/Account/Billing/Alert.tsx

```diff
@@ -245,6 +260,23 @@ const BillingTypeAlert: React.FC<BillingTypeAlertProps> = ({
   ctaHref = "/dashboard/settings/billing",
   label = "addPaymentAlert",
 }) => {
+  const [paymentMethodSaving, setPaymentMethodSaving] = useState(false);
```

Comment:
this has some code-smell:

most likely we should be moving whatever async action is happening that sets this state to this component and passing the mutation function down instead of doing the async work "below" and bubbling this state back up

if this is easy to do we should do it now, otherwise it is fine to ignore for now but we will have to revisit this.

Reply:
Good point, we can possibly revisit setting this field won in the ManageBillingButton component and make the field as optional to override